### ISSUE TITLE
Integration Tests are hanging = no releases possible currently

### ISSUE THREAD

**Issue by SebastianSchuetze** [DarqueWarrior/vsteam#424]
There seems to be a problem with the Test Stage. Artifacts cannot be downloaded, and new release would be blocked. Working on it currently, but I don't exactly know what the problem is.

Example: https://loecda.visualstudio.com/Team%20Module/_build/results?buildId=2612&view=logs&j=05c5a9c3-2da3-52a6-f69b-c4ac425f9d8a&t=46d9c746-5393-5699-861f-09f1563ec7d5

Funny thing is. If I used the same scripts locally that fail, there or recreate the pipeline in my own organization. It does not cause any problems...

**Comment by brittandeyoung** [comment]
I may be able to take a look at this. Is there a reason you are publishing the nuget package to a feed for this kind of testing instead of just using a pipeline artifact? @SebastianSchuetze

**Comment by SebastianSchuetze** [comment]
It is @DarqueWarrior idea. The general idea was to simulate the PowerShell gallery installation with "Install-Module" it would be good to simulate that as well. We had at some point problems with some users when they installed it from PS Gallery and we wanted to test this to not deploy something that does not work.

We started even to migrate the pipeline to GitHub Actions. But in Github actions we don't have a nuget feed V2 anymore which is needed for PS Module feeds.

So I figured two options to mitigate that:
1. use a container lokaly hosting such a simple feed for a short instalation test
2. use a local folder feed (via registering a folder as a package feed) which is not the same as having a real feed

Long story short. Either you 
* try it to analyze wiht Azure Pipelines. Which may be a bit harder because you have to set up everything on AzDO on your site
* or you continue the GitHub Actions implementation in the branch [actions](https://github.com/MethodsAndPractices/vsteam/tree/actions/.github/workflows)

But either way. If you can solve it, it would be of great help!!

**Comment by brittandeyoung** [comment]
I made the following PR to your pipeline hotfix branch: https://github.com/MethodsAndPractices/vsteam/pull/427

This migrates the installation of the module to use the artifact already in the pipeline. This will only affect the integration tests. 

But now that I see you Would prefer to use a nuget feed for testing purposes. I will see what I can do on the `actions` branch. I will be able to do testing on this branch?

**Comment by SebastianSchuetze** [comment]
When you make a fork you should be able to run everything with your repo. You just need to recreate some of the resources like environments. 
As far as it comes to the integration test you could just integrate those for the Cloud Version. But if you come as far as using a local nugget server feed that would already help.

**Comment by brittandeyoung** [comment]
@SebastianSchuetze Here is as far as I am able to take it without spinning up an Azure DevOps organization. I Also had to resort to using a local Nuget feed as I did not have the credentials for the GitHub Project feed.

I also found the logic for setting the build number in the semantic version was appending this number instead of properly updating the existing build number. I fixed this to update the build number.

https://github.com/MethodsAndPractices/vsteam/pull/429

Please let me know what you think.

Link to actions run getting past the Module Install: https://github.com/brittandeyoung/vsteam/actions/runs/1485308715

**Comment by SebastianSchuetze** [comment]
@brittandeyoung thanks for your help again. I was able to run integration tests successfully. There where numerous bigs somewhere in the pipeline itself (before).

So really thank you for helping:
https://github.com/MethodsAndPractices/vsteam/actions/runs/1512473303

Another questionm. Do you have an idea to simplify the script? We are downloading the dependecies as well and I was thinking if we can prevent that (e.g. when publishing the nuget package as a module.

**Comment by SebastianSchuetze** [comment]
![image](https://user-images.githubusercontent.com/7256168/143764825-2955b2ea-f111-43ac-bf7b-5c60a2e5a2fe.png)

Just have to publish to PS gallery now.
